lib/privatemessages/onMessage.js

```diff
@@ -34,19 +34,20 @@ exports.func = function (args) {
   const onMessage = new events.EventEmitter()
   let waitingForRequest = false
   let latest
-  getMessages({ jar, messageTab: 'Inbox', pageNumber: 0 })
+  getMessages({ jar: jar, messageTab: 'Inbox', pageNumber: 0, pageSize: 1 })
```

Comment:
How does getMessages work if there have been multiple messages in-between calls?
I.e.
- The event polls for messages and finds none
- User receives two messages X, Y
- Which is returned? The more recent? Does this result in some messages being missed?

Reply:
It will send the Most recent message every time a Message Notification is received.

Reply:
However, I didn't change the getMessages file so if you use getMessages separately it will show the default pageSize. pageSize is only declared in the onMessage event so each time you get a new message it will send the most recent.

Reply:
I think there is potential for this implementation to miss messages and emit the same message twice.
The message notification/event will not be fired immediately (or may not be) so it is possible to get an event and then the latest message is a different one.

I.e.
1: Message/Notification received
2: Noblox starts HTTP request
3: Another message arrives
4: Roblox responds to request with the second message

Thus, a message is lost. This is likely to be an edge case and only occur in unlucky timing scenarios, but it is possible.
Is there any data supplied with the message notification that would allow the API to return the correct message for that notification? Otherwise the best approach would be to loop through and return all, storing the latest message returned (as the current API does)

Reply:
I will be doing testing over the next few days to make sure it works like it did before I made the change. When I did the change, I thought it would send every single message so I will check it out.